beacon_node/src/config.rs

```diff
@@ -468,9 +468,30 @@ pub fn get_config<E: EthSpec>(
     client_config.chain.checkpoint_sync_url_timeout =
         clap_utils::parse_required::<u64>(cli_args, "checkpoint-sync-url-timeout")?;
 
-    client_config.genesis = if let Some(genesis_state_bytes) =
-        eth2_network_config.genesis_state_bytes.clone()
-    {
+    client_config.genesis_state_url_timeout =
+        clap_utils::parse_required(cli_args, "genesis-state-url-timeout")
+            .map(Duration::from_secs)?;
+
+    let genesis_state_url_opt =
+        clap_utils::parse_optional::<String>(cli_args, "genesis-state-url")?;
+    let checkpoint_sync_url_opt =
+        clap_utils::parse_optional::<String>(cli_args, "checkpoint-sync-url")?;
+
+    // If the `--genesis-state-url` is defined, use that to download the
+    // genesis state bytes. If it's not defined, try `--checkpoint-sync-url`.
+    client_config.genesis_state_url = if let Some(genesis_state_url) = genesis_state_url_opt {
+        Some(genesis_state_url)
```

Comment:
Does `genesis_state_url` actually get used if `--checkpoint-sync-url` is supplied? Wouldn't it just go with checkpoint sync because we get `ClientGenesis::CheckpointSyncUrl` instead of `ClientGenesis::GenesisState` below?

https://github.com/sigp/lighthouse/blob/766a1c360d25698682d9f59e5c448ace300b7078/beacon_node/src/config.rs#L519-L526

Reply:
We still fetch the genesis state when doing a checkpoint sync: https://github.com/paulhauner/lighthouse/blob/d18edb7c51ec03e903dbff4148255bdddd6806e0/beacon_node/client/src/builder.rs#L380

I have a note in the PR description that talks about this too:

> When we do a fresh checkpoint sync we're downloading the genesis state to check the genesis_validators_root against the finalised state we receive. This is not entirely pointless, since we verify the checksum when we download the genesis state so we are actually guaranteeing that the finalised state is on the same network. There might be a smarter/less-download-y way to go about this, but I've run out of cycles to figure that out. Perhaps we can grab it in the next release?

Reply:
Ah I see, I missed that. Thanks!